src/PoolManager.sol

```diff
@@ -360,7 +317,11 @@ contract PoolManager is Auth, IPoolManager {
 
         // Give investment manager infinite approval for asset
         // in the escrow to transfer to the user on redeem or withdraw
-        escrow.approveMax(asset, address(investmentManager));
+        escrow.approveMax(asset, investmentManager);
+
+        // Give pool manager infinite approval for asset
+        // in the escrow to transfer to the user on transfer
+        escrow.approveMax(asset, address(this));
```

Comment:
This was in `handleTransfer` every time, it's more gas efficient to just do it once.